lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -7038,32 +7038,84 @@ class DecomposeAtenAdaptiveAvgPool2dOp
     Value constantZero = rewriter.create<Torch::ConstantIntOp>(
         loc, rewriter.getI64IntegerAttr(0));
+    Value constantOne = rewriter.create<Torch::ConstantIntOp>(
+        loc, rewriter.getI64IntegerAttr(1));
+    Value constantTwo = rewriter.create<Torch::ConstantIntOp>(
+        loc, rewriter.getI64IntegerAttr(2));
     Value constantFalse = rewriter.create<Torch::ConstantBoolOp>(loc, false);
     Value constantTrue = rewriter.create<Torch::ConstantBoolOp>(loc, true);
     Value constantNone = rewriter.create<Torch::ConstantNoneOp>(loc);
-    SmallVector<Value, 2> kernelSize;
 
+    SmallVector<Value, 2> strideSize;
+    SmallVector<Value, 2> kernelSize;
     for (unsigned i = 0; i < inputHW.size(); i++) {
       Value remainder = rewriter.create<AtenRemainderIntOp>(
           loc, inputHW[i], outputShapeSizesTorchInt[i]);
-      Value cond = rewriter.create<AtenEqIntOp>(loc, remainder, constantZero);
-      rewriter.create<RuntimeAssertOp>(loc, cond,
-                                       "unimplemented: only support cases "
-                                       "input size is an integer multiple of "
-                                       "output size");
-      Value stride = rewriter.create<AtenFloordivIntOp>(
+
+      // Filter cases with fixed stride size.
+      Value cond1 = rewriter.create<Torch::AtenGtIntOp>(
+          loc, rewriter.getType<Torch::BoolType>(), outputShapeSizesTorchInt[i],
+          rewriter.create<Torch::AtenMulIntOp>(
+              loc, rewriter.getType<Torch::IntType>(), remainder,
+              rewriter.create<Torch::AtenSubIntOp>(
+                  loc, outputShapeSizesTorchInt[i], constantOne)));
+
+      // Filter cases with fixed kernel size.
+      Value cond2 =
+          rewriter.create<Torch::AtenEqIntOp>(loc, remainder, constantZero);
+      cond2 = rewriter.create<Torch::AtenIntBoolOp>(loc, cond2);
+      Value offset = rewriter.create<Torch::AtenIntBoolOp>(
+          loc, rewriter.create<Torch::Aten__Not__Op>(
+                   loc, rewriter.create<Torch::AtenBoolIntOp>(loc, remainder)));
+      Value remainder_not_zero =
+          rewriter.create<Torch::AtenAddIntOp>(loc, remainder, offset);
+      Value cond3 = rewriter.create<Torch::AtenEqIntOp>(
+          loc,
+          rewriter.create<Torch::AtenRemainderIntOp>(
+              loc, outputShapeSizesTorchInt[i], remainder_not_zero),
+          constantZero);
+      cond3 = rewriter.create<Torch::AtenIntBoolOp>(loc, cond3);
+      Value cond = rewriter.create<Torch::AtenBoolIntOp>(
+          loc, rewriter.create<Torch::AtenAddIntOp>(loc, cond2, cond3));
+
+      Value condAnd = rewriter.create<Torch::AtenAddIntOp>(
+          loc, rewriter.create<Torch::AtenIntBoolOp>(loc, cond1),
+          rewriter.create<Torch::AtenIntBoolOp>(loc, cond));
+      Value finalCond =
+          rewriter.create<Torch::AtenEqIntOp>(loc, condAnd, constantTwo);
+
+      rewriter.create<RuntimeAssertOp>(
```

Comment:
It's a little coupled. I think it's more clear to split `RuntimeAssert`, one for kernel and another for stride.
BTW, the code above is hard for human reading. Could you make it closer to the formula described?